packages/dds/tree/src/test/feature-libraries/editable-tree/editableTree.editing.spec.ts

```diff
@@ -0,0 +1,287 @@
+                    singleTextCursor({ type: stringSchema.name, value: "second" }),
+                    singleTextCursor({ type: stringSchema.name, value: "first" }),
+                ].forEach((content) => field_0.insertNodes(0, content));
+                assert.throws(
+                    () => field_0.insertNodes(5, singleTextCursor({ type: stringSchema.name })),
+                    (e) => validateAssertionError(e, "Index must be less than or equal to length."),
+                    "Expected exception was not thrown",
+                );
+                assert.equal(field_0[0], "first");
+                assert.equal(field_0[1], "second");
+                await provider.ensureSynchronized();
+                assert.deepEqual(field_0, field_1);
+
+                // delete using `deleteNodes()`
+                field_0.deleteNodes(1, 1);
+                assert.throws(
+                    () => field_0.deleteNodes(2),
+                    (e) => validateAssertionError(e, "Index must be less than length."),
+                    "Expected exception was not thrown",
+                );
+                assert.equal(field_0.length, 2);
+                assert.throws(
+                    () => field_0.deleteNodes(0, -1),
+                    (e) => validateAssertionError(e, "Count must be non-negative."),
+                    "Expected exception was not thrown",
+                );
+                await provider.ensureSynchronized();
+                assert.deepEqual(field_0, field_1);
+                field_0.deleteNodes(0, 5);
+                assert.equal(field_0.length, 0);
+                assert.doesNotThrow(() => field_0.deleteNodes(0, 0));
+                await provider.ensureSynchronized();
+                assert.deepEqual(field_0, field_1);
+
+                trees[0].context.free();
+                trees[1].context.free();
+            });
+
+            it("as optional field", async () => {
+                const [provider, trees] = await createSharedTrees(
+                    getTestSchema(FieldKinds.optional),
+                    [{ type: rootSchemaName }],
+                    2,
+                );
+                assert(isUnwrappedNode(trees[0].root));
+                assert(isUnwrappedNode(trees[1].root));
+
+                // create
+                trees[0].root[createFieldSymbol](
+                    fieldKey,
+                    singleTextCursor({ type: stringSchema.name, value: "foo" }),
+                );
+                await provider.ensureSynchronized();
+                assert.equal(trees[1].root[fieldKey], "foo");
+
+                // edit using assignment
+                trees[0].root[fieldKey] = "bar";
+                await provider.ensureSynchronized();
+                assert.equal(trees[0].root[fieldKey], "bar");
+
+                // edit using valueSymbol
+                trees[0].root[getWithoutUnwrappingSymbol](fieldKey).getWithoutUnwrapping(0)[
```

Comment:
I thinks this is the most "user unfriendly" case, as the user must always follow on what is a field and what is a node. On the other hand, as currently `valueSymbol` makes sense only for primitives, I can hardly imagine a usecase where it will be better to use it instead of just a simple assignment.

Reply:
Maybe, it might become more user friendly, if we rename `getWithoutUnwrapping` to `getField` and `getNode` and hint on "without unwrapping" by return type of those?

Reply:
valueSymbol should work for any node with a value (which might not be a primitive if the value is ValueSchema.Seralizable, or if the node has both a value and fields). 

You also need to use valueSymbol instead of assignment in code that does not have access to the parent node or field.

Renaming the symbols seems like good idea: I think your suggested names are an improvement.

As long as it doesn't cause too many issues with name collisions for imports/exports, we can leave "Symbol" out of the names. Removing `WithoutUnwrapping` also seems like a good idea. Instead, we could call versions which do unwrapping (if we have any which actually have named methods) something like getUnwrapped. Another option is to just replace `WithoutUnwrapping` with `Wrapped`, but I think your approach of not saying anything in that case is better.

Reply:
@CraigMacomber Alright, I have changed the names for `getWithoutUnwrappingSymbol` and `createFieldSymbol` to be just `getField` and `createField`, and `getWithoutUnwrapping` to `getNode`.
I kept `Symbol` for value, type and type name since it's very probable that they'll overlap with variable names, and thus will bring more headache rather than a convenience out of omitting the `Symbol` part, whereas `getField` and `createField` are conventionally function or even method names, for which the name clashing should not occur so often, in my opinion.